gf-webapp-admin/src/component/filter/RoomsBlockWithFilters.tsx

```diff
@@ -0,0 +1,91 @@
+import {Room} from "src/model/Room/Room";
+import {useFilterRooms} from "src/component/filter/FilterContext";
+import {RoomSimplified} from "src/component/rooms/roomSimplified/RoomSimplified";
+import {getDateValue} from "src/utils/getDateValue";
+import styles from "src/component/filter/Filters.module.scss";
+
+interface RoomsListProps {
+  rooms:Room[]
+}
+
+export const RoomsBlockWithFilters = (props: RoomsListProps) => {
+  const {searchValue, adultsValue, childrenValue, dateInValue, dateOutValue} = useFilterRooms();
+
+  const filterAdults = (room: Room) => {
+    if (adultsValue !== 0) {
+      return room.adults === adultsValue;
+    } else {
+      return room;
+    }
+  };
+
+  const filterChildren = (room: Room) => {
+    if (childrenValue !== 0) {
+      return room.children === childrenValue;
+    } else {
+      return room;
+    }
+  };
+
+  const filterSearch = (room: Room) => {
+    return Object.values(room)
+      .join()
+      .toString()
+      .toLowerCase()
+      .includes(searchValue);
+  };
+
+  const filterRoomsNotBooked = (room: Room) => {
+    return room.booked
+      .filter(roomBooked =>
+        (getDateValue(roomBooked.period.dateOut) < getDateValue(dateInValue)
+    && getDateValue(roomBooked.period.dateOut) < getDateValue(dateOutValue)) ||
+    (getDateValue(roomBooked.period.dateIn) > getDateValue(dateInValue)
+        && getDateValue(roomBooked.period.dateIn) > getDateValue(dateOutValue)));
+  };
+
+  const dateInAfterDateOut = getDateValue(dateInValue) >= getDateValue(dateOutValue);
+  const dateInBeforeToday = getDateValue(dateInValue) < (getDateValue() - 100000000);
+  const wrongDates = dateInAfterDateOut || dateInBeforeToday;
+
+  const filterDate = (room: Room) => {
+    if (wrongDates) {
```

Comment:
You can disable the option to select a past date in the calendar.
And after selecting the check-in date, disable the option in the check-out date calendar to select a date earlier than the check-in date. If you do this, you don't need to check that the arrival date is later than the departure date.
Now when selecting the check-in date, an alert immediately appears because the check-out date is earlier than the check-in date

Reply:
Done

Reply:
Good.
But it is a little strange that once a check-out date is selected, all days before the selected one become unavailable, even though they are later than the check-in date

Reply:
Corrected